### DISCUSSION TITLE
Is it possible to use raw strings without `sql` templating?

### DISCUSSION THREAD

**Discussion by dhruv-nudge** [porsager/postgres#989] [General]
I am having my own query mapper layer which uses string templating, and it is conflicting with string templating of postgres.js in a weird way. For e.g. I am not able to pass order by column simply like `param.orderby` in the raw SQL.

**Comment by porsager** [comment]
https://github.com/porsager/postgres?tab=readme-ov-file#unsafe-raw-string-queries

**Reply by Dhruv-Garg79** [reply]
Hey @porsager, I checked the difference between both approaches and wanted to confirm if we still get protection from SQL injection if we pass values as parameters in unsafe calls. e.g.

```
await sql.unsafe('select * from nudge.user where id = $1', ["307edd8e-1179-4f51-90f8-7cd58ce50719'--"]),
```

I checked same call with recommended templating way, but that also didn't threw any error for this particular SQL injection example

```
const where = "307edd8e-1179-4f51-90f8-7cd58ce50719'--";
await sql`select * from nudge.user where id = ${where}`,
```
